test/powershell/engine/Job/Jobs.Tests.ps1

```diff
@@ -1,23 +1,275 @@
+
+        It 'Can wait for any job to complete' {
+            $jobs = 1..3 | ForEach-Object { $seconds = $_ ; Start-Job -ScriptBlock { Start-Sleep -Seconds $using:seconds ; $using:seconds} }
+            $waitedJob = Wait-Job -Job $jobs -Any
+            ValidateJobInfo -job $waitedJob -state 'Completed' -hasMoreData $true -command ' Start-Sleep -Seconds $using:seconds ; $using:seconds'
+            $result = $waitedJob | Receive-Job
+            ## We check for $result to be less than 4 so that any of the jobs completing first will considered a success.
+            $result | Should BeLessThan 4
+            $jobs | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can timeout waiting for a job' {
+            $job = Start-Job -ScriptBlock { Start-Sleep -Seconds 10 }
+            $job | Wait-Job -TimeoutSec 1
+            ValidateJobInfo -job $job -state 'Running' -hasMoreData $true -command ' Start-Sleep -Seconds 10 '
+            $job | Remove-Job -Force
+        }
     }
 
-    AfterAll {
-        Remove-Job $job -Force
+    Context 'Receive-job tests' {
+        It   'Can receive-job with state change events' {
+            $result = Start-Job -Name 'ReceiveWriteEventsJob' -ScriptBlock { 1 + 1 } | Receive-Job -Wait -WriteEvents
+            $result.Count | Should Be 3
+            $result[0] | Should Be 2
+            $result[1].GetType().FullName | Should BeExactly 'System.Management.Automation.JobStateEventArgs'
+            Remove-Job -Name 'ReceiveWriteEventsJob' -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job with job object and result' {
+            $result = Start-Job -ScriptBlock { 1 + 1 } | Receive-Job -Wait -WriteJobInResults
+            $result.Count | Should Be 2
+            ValidateJobInfo -job $result[0] -command ' 1 + 1 ' -state 'Completed' -hasMoreData $false
+            $result[1] | Should Be 2
+            $result[0] | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job and autoremove' {
+            $result = Start-Job -Name 'ReceiveJobAutoRemove' -ScriptBlock { 1 + 1 } | Receive-Job -Wait -AutoRemoveJob
+            $result | Should Be 2
+            { Get-Job -Name 'ReceiveJobAutoRemove' -ErrorAction Stop } | ShouldBeErrorId 'JobWithSpecifiedNameNotFound,Microsoft.PowerShell.Commands.GetJobCommand'
+        }
+
+        It 'Can receive-job and keep results' {
+            $job = Start-Job -ScriptBlock { 1 + 1 } | Wait-Job
+            $result = Receive-Job -Keep -Job $job
+            $result | Should Be 2
+            $result2 = Receive-Job -Job $job
+            $result2 | Should Be 2
+            $result3 = Receive-Job -Job $job
+            $result3 | Should BeNullOrEmpty
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job with NoRecurse' {
+            $job = Start-Job -ScriptBlock { 1 + 1 }
+            $result = Receive-Job -Wait -NoRecurse -Job $job
```

Comment:
This test doesn't actually test the NoRecurse functionality. There should be a child job and a check should be made to ensure that it gets the correct value.

Reply:
All jobs create a childjob which returns the result. Hence with NoRecurse we cannot get the result.